ui/apps/platform/src/Containers/Collections/CollectionsFormPage.tsx

```diff
@@ -85,6 +85,9 @@ function CollectionsFormPage({
     } else if (loading) {
         content = <>{/* TODO - Handle UI for loading state */}</>;
     } else if (initialData) {
+        if (pageAction.type === 'clone') {
+            initialData.name += ' (COPY)';
+        }
```

Comment:
is that although control flow prevents `initialData` from referring to `defaultCollectionData` for clone action, therefore cannot mutate it,

A question for thought (not blocker to merge) is whether there is an alternative control flow which would allow TypeScript to report such a mutation as a problem? Maybe a prior question is does TypeScript report a problem with current code if `defaultCollectionData` has `as const` annotation?

Reply:
Commenting here because I noticed while reading code, but it is hard to comment on the line above:

```js
{getAxiosErrorMessage(error)}
```

Less important now that you implemented permission checks, but allows any other generic logic.

Reply:
I thought about making the `defaultCollectionData` `as const` too, but didn't since it would require the `Collection` type to use `readonly` for the `embeddedCollectionIds` and `resourceSelector` properties. It is something I thought about when working on the dashboards too, but didn't consider it too strongly since it didn't seem to be a pattern that was used extensively elsewhere.

I think I generally just assume that the data structures I use are immutable even if it isn't strictly enforced. (And sometimes take advantage over it not being so strict like in this case.)

I wonder how Formik would handle things if we passed it immutable initial data? A quick look at their source shows they are running state through a reducer so chances are it would work out fine. I can experiment with this in a follow up.